Doesn't work with L1 RPC Geth

#### System information

zkEVM Node version: `v0.0.3-RC36`
OS & Version: Ubuntu 22.04.1 LTS
Network: Mainnet

#### Steps to reproduce the behaviour

I've tried using multiple Geth archive nodes as L1 RPC but they all crash with the following errors. Nethermind and Erigon works fine without any issues. 

#### Logs

````
Apr 06 20:13:28  zkevm-node[30287]: 2023-04-06T20:13:28.564Z        FATAL        cmd/run.go:98        error getting forks: request timed out
Apr 06 20:13:28  zkevm-node[30287]: /tmp/zkevmnode-0.0.3-RC36/log/log.go:140 github.com/0xPolygonHermez/zkevm-node/log.appendStackTraceMaybeArgs()
Apr 06 20:13:28  zkevm-node[30287]: /tmp/zkevmnode-0.0.3-RC36/log/log.go:221 github.com/0xPolygonHermez/zkevm-node/log.Fatal()
Apr 06 20:13:28  zkevm-node[30287]: /tmp/zkevmnode-0.0.3-RC36/cmd/run.go:98 main.start()
Apr 06 20:13:28  zkevm-node[30287]: /home/ansible-10000/go/pkg/mod/github.com/urfave/cli/v2@v2.25.1/command.go:274 github.com/urfave/cli/v2.(*Command).Run()
Apr 06 20:13:28  zkevm-node[30287]: /home/ansible-10000/go/pkg/mod/github.com/urfave/cli/v2@v2.25.1/command.go:267 github.com/urfave/cli/v2.(*Command).Run()
Apr 06 20:13:28  zkevm-node[30287]: /home/ansible-10000/go/pkg/mod/github.com/urfave/cli/v2@v2.25.1/app.go:332 github.com/urfave/cli/v2.(*App).RunContext()
Apr 06 20:13:28  zkevm-node[30287]: /home/ansible-10000/go/pkg/mod/github.com/urfave/cli/v2@v2.25.1/app.go:309 github.com/urfave/cli/v2.(*App).Run()
Apr 06 20:13:28  zkevm-node[30287]: /tmp/zkevmnode-0.0.3-RC36/cmd/main.go:140 main.main()
Apr 06 20:13:28  zkevm-node[30287]: /usr/local/go/src/runtime/proc.go:250 runtime.main()
Apr 06 20:13:28  zkevm-node[30287]:         {"pid": 30287, "version": "v0.0.3-RC36"}
Apr 06 20:13:28  zkevm-node[30287]: main.start
Apr 06 20:13:28  zkevm-node[30287]:         /tmp/zkevmnode-0.0.3-RC36/cmd/run.go:98
Apr 06 20:13:28  zkevm-node[30287]: github.com/urfave/cli/v2.(*Command).Run
Apr 06 20:13:28  zkevm-node[30287]:         /home/ansible-10000/go/pkg/mod/github.com/urfave/cli/v2@v2.25.1/command.go:274
Apr 06 20:13:28  zkevm-node[30287]: github.com/urfave/cli/v2.(*Command).Run
Apr 06 20:13:28  zkevm-node[30287]:         /home/ansible-10000/go/pkg/mod/github.com/urfave/cli/v2@v2.25.1/command.go:267
Apr 06 20:13:28  zkevm-node[30287]: github.com/urfave/cli/v2.(*App).RunContext
Apr 06 20:13:28  zkevm-node[30287]:         /home/ansible-10000/go/pkg/mod/github.com/urfave/cli/v2@v2.25.1/app.go:332
Apr 06 20:13:28  zkevm-node[30287]: github.com/urfave/cli/v2.(*App).Run
Apr 06 20:13:28  zkevm-node[30287]:         /home/ansible-10000/go/pkg/mod/github.com/urfave/cli/v2@v2.25.1/app.go:309
Apr 06 20:13:28  zkevm-node[30287]: main.main
Apr 06 20:13:28  zkevm-node[30287]:         /tmp/zkevmnode-0.0.3-RC36/cmd/main.go:140
Apr 06 20:13:28  zkevm-node[30287]: runtime.main
Apr 06 20:13:28  zkevm-node[30287]:         /usr/local/go/src/runtime/proc.go:250
Apr 06 20:13:28  systemd[1]: zksync.service: Main process exited, code=exited, status=1/FAILURE
````

--- a/cmd/run.go
+++ b/cmd/run.go
@@ -119,9 +119,12 @@ func start(cliCtx *cli.Context) error {
 	}
 	// Read Fork ID FROM POE SC
 	forkIDIntervals, err := etherman.GetForks(cliCtx.Context, c.NetworkConfig.Genesis.GenesisBlockNum)
-	if err != nil || len(forkIDIntervals) == 0 {
-		log.Fatal("error getting forks: ", err)
+	if err != nil {
+		log.Fatal("error getting forks. Please check the configuration. Error: ", err)
+	} else if len(forkIDIntervals) == 0 {
+		log.Fatal("error: no forkID received. It should receive at least one, please check the configuration...")
 	}
+
 	currentForkID := forkIDIntervals[len(forkIDIntervals)-1].ForkId
 
 	c.Aggregator.ChainID = l2ChainID

--- a/cmd/run.go
+++ b/cmd/run.go
@@ -118,7 +118,7 @@ func start(cliCtx *cli.Context) error {
 		log.Fatal(err)
 	}
 	// Read Fork ID FROM POE SC
-	forkIDIntervals, err := etherman.GetForks(cliCtx.Context)
+	forkIDIntervals, err := etherman.GetForks(cliCtx.Context, c.NetworkConfig.Genesis.GenesisBlockNum)
 	if err != nil || len(forkIDIntervals) == 0 {
 		log.Fatal("error getting forks: ", err)
 	}

--- a/config/environments/local/local.genesis.config.json
+++ b/config/environments/local/local.genesis.config.json
@@ -6,7 +6,7 @@
 		"polygonZkEVMGlobalExitRootAddress": "0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6"
 	},
   "root": "0xd88680f1b151dd67518f9aca85161424c0cac61df2f5424a3ddc04ea25adecc7",
-  "genesisBlockNumber": 74,
+  "genesisBlockNumber": 65,
   "genesis": [
    {
     "contractName": "PolygonZkEVMDeployer",

--- a/db/scripts/init_prover_db.sql
+++ b/db/scripts/init_prover_db.sql
@@ -8,6 +8,7 @@ CREATE TABLE state.program (hash BYTEA PRIMARY KEY, data BYTEA NOT NULL);
 
 CREATE USER prover_user with password 'prover_pass';
 GRANT CONNECT ON DATABASE prover_db TO prover_user;
+ALTER USER prover_user SET SEARCH_PATH=state;
 GRANT ALL PRIVILEGES ON SCHEMA state TO prover_user;
 GRANT ALL PRIVILEGES ON TABLE state.nodes TO prover_user;
 GRANT ALL PRIVILEGES ON TABLE state.program TO prover_user;

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -59,7 +59,7 @@ services:
     restart: unless-stopped
     image: postgres
     healthcheck:
-      test: [ "CMD-SHELL", "pg_isready" ]
+      test: ["CMD-SHELL", "pg_isready -d $${POSTGRES_DB} -U $${POSTGRES_USER}"]
       interval: 10s
       timeout: 5s
       retries: 5
@@ -85,7 +85,7 @@ services:
     restart: unless-stopped
     image: postgres
     healthcheck:
-      test: [ "CMD-SHELL", "pg_isready" ]
+      test: ["CMD-SHELL", "pg_isready -d $${POSTGRES_DB} -U $${POSTGRES_USER}"]
       interval: 10s
       timeout: 5s
       retries: 5

--- a/docs/production-setup.md
+++ b/docs/production-setup.md
@@ -67,7 +67,7 @@ If everything has gone as expected you should be able to run queries to the JSON
 ## Stop
 
 ```bash
-sudo docker compose --env-file $ZKEVM_CONFIG_DIR/.env -f $ZKEVM_DIR/$ZKEVM_NET/docker-compose.yml down
+docker compose --env-file $ZKEVM_CONFIG_DIR/.env -f $ZKEVM_DIR/$ZKEVM_NET/docker-compose.yml down
 ```
 
 ## Updating

--- a/etherman/etherman.go
+++ b/etherman/etherman.go
@@ -2,7 +2,6 @@ package etherman
 
 import (
 	"context"
-	"encoding/hex"
 	"encoding/json"
 	"errors"
 	"fmt"
@@ -192,35 +191,36 @@ func NewClient(cfg Config, l1Config L1Config) (*Client, error) {
 
 // VerifyGenBlockNumber verifies if the genesis Block Number is valid
 func (etherMan *Client) VerifyGenBlockNumber(ctx context.Context, genBlockNumber uint64) (bool, error) {
-	genBlock := big.NewInt(0).SetUint64(genBlockNumber)
-	response, err := etherMan.EthClient.CodeAt(ctx, etherMan.l1Cfg.ZkEVMAddr, genBlock)
+	log.Info("Verifying genesis blockNumber: ", genBlockNumber)
+	// Filter query
+	genBlock := new(big.Int).SetUint64(genBlockNumber)
+	query := ethereum.FilterQuery{
+		FromBlock: genBlock,
+		ToBlock: genBlock,
+		Addresses: etherMan.SCAddresses,
+		Topics:    [][]common.Hash{{updateZkEVMVersionSignatureHash}},
+	}
+	logs, err := etherMan.EthClient.FilterLogs(ctx, query)
 	if err != nil {
-		log.Error("error getting smc code for gen block number. Error: ", err)
 		return false, err
 	}
-	responseString := hex.EncodeToString(response)
-	if responseString == "" {
-		return false, nil
+	if len(logs) == 0 {
+		return false, fmt.Errorf("the specified genBlockNumber in config file does not contain any forkID event. Please use the proper blockNumber.")
 	}
-	responsePrev, err := etherMan.EthClient.CodeAt(ctx, etherMan.l1Cfg.ZkEVMAddr, genBlock.Sub(genBlock, big.NewInt(1)))
+	zkevmVersion, err := etherMan.ZkEVM.ParseUpdateZkEVMVersion(logs[0])
 	if err != nil {
-		if parsedErr, ok := tryParseError(err); ok {
-			if errors.Is(parsedErr, ErrMissingTrieNode) {
-				return true, nil
-			}
-		}
-		log.Error("error getting smc code for gen block number. Error: ", err)
+		log.Error("error parsing the forkID event")
 		return false, err
 	}
-	responsePrevString := hex.EncodeToString(responsePrev)
-	if responsePrevString != "" {
-		return false, nil
+	if zkevmVersion.NumBatch != 0 {
+		return false, fmt.Errorf("the specified genBlockNumber in config file does not contain the initial forkID event (BatchNum: %d). Please use the proper blockNumber.", zkevmVersion.NumBatch)
 	}
 	return true, nil
 }
 
 // GetForks returns fork information
 func (etherMan *Client) GetForks(ctx context.Context, genBlockNumber uint64) ([]state.ForkIDInterval, error) {
+	log.Debug("Getting forkIDs from blockNumber: ", genBlockNumber)
 	// Filter query
 	query := ethereum.FilterQuery{
 		FromBlock: new(big.Int).SetUint64(genBlockNumber),

--- a/etherman/etherman.go
+++ b/etherman/etherman.go
@@ -220,10 +220,10 @@ func (etherMan *Client) VerifyGenBlockNumber(ctx context.Context, genBlockNumber
 }
 
 // GetForks returns fork information
-func (etherMan *Client) GetForks(ctx context.Context) ([]state.ForkIDInterval, error) {
+func (etherMan *Client) GetForks(ctx context.Context, genBlockNumber uint64) ([]state.ForkIDInterval, error) {
 	// Filter query
 	query := ethereum.FilterQuery{
-		FromBlock: new(big.Int).SetUint64(1),
+		FromBlock: new(big.Int).SetUint64(genBlockNumber),
 		Addresses: etherMan.SCAddresses,
 		Topics:    [][]common.Hash{{updateZkEVMVersionSignatureHash}},
 	}

--- a/etherman/etherman.go
+++ b/etherman/etherman.go
@@ -196,7 +196,7 @@ func (etherMan *Client) VerifyGenBlockNumber(ctx context.Context, genBlockNumber
 	genBlock := new(big.Int).SetUint64(genBlockNumber)
 	query := ethereum.FilterQuery{
 		FromBlock: genBlock,
-		ToBlock: genBlock,
+		ToBlock:   genBlock,
 		Addresses: etherMan.SCAddresses,
 		Topics:    [][]common.Hash{{updateZkEVMVersionSignatureHash}},
 	}

--- a/etherman/etherman_test.go
+++ b/etherman/etherman_test.go
@@ -398,7 +398,7 @@ func TestGetForks(t *testing.T) {
 	// Set up testing environment
 	etherman, _, _, _, _ := newTestingEnv()
 	ctx := context.Background()
-	forks, err := etherman.GetForks(ctx)
+	forks, err := etherman.GetForks(ctx, 0)
 	require.NoError(t, err)
 	assert.Equal(t, 1, len(forks))
 	assert.Equal(t, uint64(1), forks[0].ForkId)

--- a/synchronizer/interfaces.go
+++ b/synchronizer/interfaces.go
@@ -22,7 +22,7 @@ type ethermanInterface interface {
 	GetLatestBatchNumber() (uint64, error)
 	GetTrustedSequencerURL() (string, error)
 	VerifyGenBlockNumber(ctx context.Context, genBlockNumber uint64) (bool, error)
-	GetForks(ctx context.Context) ([]state.ForkIDInterval, error)
+	GetForks(ctx context.Context, genBlockNumber uint64) ([]state.ForkIDInterval, error)
 }
 
 // stateInterface gathers the methods required to interact with the state.

--- a/synchronizer/mock_etherman.go
+++ b/synchronizer/mock_etherman.go
@@ -48,25 +48,25 @@ func (_m *ethermanMock) EthBlockByNumber(ctx context.Context, blockNumber uint64
 	return r0, r1
 }
 
-// GetForks provides a mock function with given fields: ctx
-func (_m *ethermanMock) GetForks(ctx context.Context) ([]state.ForkIDInterval, error) {
-	ret := _m.Called(ctx)
+// GetForks provides a mock function with given fields: ctx, genBlockNumber
+func (_m *ethermanMock) GetForks(ctx context.Context, genBlockNumber uint64) ([]state.ForkIDInterval, error) {
+	ret := _m.Called(ctx, genBlockNumber)
 
 	var r0 []state.ForkIDInterval
 	var r1 error
-	if rf, ok := ret.Get(0).(func(context.Context) ([]state.ForkIDInterval, error)); ok {
-		return rf(ctx)
+	if rf, ok := ret.Get(0).(func(context.Context, uint64) ([]state.ForkIDInterval, error)); ok {
+		return rf(ctx, genBlockNumber)
 	}
-	if rf, ok := ret.Get(0).(func(context.Context) []state.ForkIDInterval); ok {
-		r0 = rf(ctx)
+	if rf, ok := ret.Get(0).(func(context.Context, uint64) []state.ForkIDInterval); ok {
+		r0 = rf(ctx, genBlockNumber)
 	} else {
 		if ret.Get(0) != nil {
 			r0 = ret.Get(0).([]state.ForkIDInterval)
 		}
 	}
 
-	if rf, ok := ret.Get(1).(func(context.Context) error); ok {
-		r1 = rf(ctx)
+	if rf, ok := ret.Get(1).(func(context.Context, uint64) error); ok {
+		r1 = rf(ctx, genBlockNumber)
 	} else {
 		r1 = ret.Error(1)
 	}

--- a/synchronizer/synchronizer.go
+++ b/synchronizer/synchronizer.go
@@ -565,7 +565,7 @@ func (s *ClientSynchronizer) processForkID(forkID etherman.ForkID, blockNumber u
 	}
 	if latestBatchNumber < forkID.BatchNumber { //If the forkID will start in a future batch
 		// Read Fork ID FROM POE SC
-		forkIDIntervals, err := s.etherMan.GetForks(s.ctx)
+		forkIDIntervals, err := s.etherMan.GetForks(s.ctx, s.genesis.GenesisBlockNum)
 		if err != nil || len(forkIDIntervals) == 0 {
 			log.Error("error getting all forkIDs: ", err)
 			rollbackErr := dbTx.Rollback(s.ctx)
@@ -597,7 +597,7 @@ func (s *ClientSynchronizer) processForkID(forkID etherman.ForkID, blockNumber u
 	}
 
 	// Read Fork ID FROM POE SC
-	forkIDIntervals, err := s.etherMan.GetForks(s.ctx)
+	forkIDIntervals, err := s.etherMan.GetForks(s.ctx, s.genesis.GenesisBlockNum)
 	if err != nil || len(forkIDIntervals) == 0 {
 		log.Error("error getting all forkIDs: ", err)
 		rollbackErr := dbTx.Rollback(s.ctx)

--- a/test/config/test.genesis.config.json
+++ b/test/config/test.genesis.config.json
@@ -6,7 +6,7 @@
 		"polygonZkEVMGlobalExitRootAddress": "0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6"
 	},
   "root": "0xd88680f1b151dd67518f9aca85161424c0cac61df2f5424a3ddc04ea25adecc7",
-  "genesisBlockNumber": 74,
+  "genesisBlockNumber": 65,
   "genesis": [
    {
     "contractName": "PolygonZkEVMDeployer",

--- a/test/scripts/sendForcedBatch/main.go
+++ b/test/scripts/sendForcedBatch/main.go
@@ -105,7 +105,7 @@ func sendForcedBatches(cliCtx *cli.Context) error {
 	log.Debug("currentBlock.Time(): ", currentBlock.Time())
 
 	// Get tip
-	tip, err := poe.BatchFee(&bind.CallOpts{Pending: false})
+	tip, err := poe.GetForcedBatchFee(&bind.CallOpts{Pending: false})
 	if err != nil {
 		log.Error("error getting tip. Error: ", err)
 		return err

